capsules/src/process_console.rs

```diff
@@ -984,127 +1157,18 @@ impl<'a, const COMMAND_HISTORY_LEN: usize, A: Alarm<'a>, C: ProcessManagementCap
                                 // Remove last byte from the command in order
                                 // not to permit accumulation of the text
                                 if COMMAND_HISTORY_LEN > 1 {
-                                    self.command_history.map(|cmd_arr| {
-                                        (&mut cmd_arr[0]).delete_last_byte();
+                                    self.command_history.map(|ht| {
+                                        ht.cmds[0].delete_last_byte();
                                     });
                                 }
                             }
-                        } else if (COMMAND_HISTORY_LEN > 1)
-                            && (read_buf[0] == ESC || self.control_seq_in_progress.get())
+                        } else if (COMMAND_HISTORY_LEN > 1) && (esc_state.has_started()) {
+                            self.command_history
+                                .map(|ht| ht.modified_byte = previous_byte);
+                        } else if index < (command.len() - 1)
+                            && read_buf[0] < 128
```

Comment:
128?

Reply:
This was not added by me, I can't say what is it used for.

Reply:
I have inspected the `File Blame` using the git lens and seen that `read_buf[0] < 128` was added by the creator of the ProcessConsole `Philips Levis` with the following explanation:

```text
// For some reason, sometimes reads return > 127 but no error,
// which causes utf-8 decoding failure, so check byte is < 128. -pal
```

However, I created a constant for 128, in order not to make further confusions.